src/main/java/ru/otus/vadim/ivanov/java/basic/lesson11/oop2/WaterfowlAnimal.java

```diff
@@ -0,0 +1,31 @@
+package ru.otus.vadim.ivanov.java.basic.lesson11.oop2;
+
+//водоплавающие
+public abstract class WaterfowlAnimal extends Animal {
+
+    private int swimCost;
+    private int swimSpeed;
+    public WaterfowlAnimal(String name, int runSpeed, int swimSpeed, int endurance) {
+        super(name,runSpeed,endurance);
+        this.swimSpeed = swimSpeed;
+
+        this.swimCost = initSwimCost();
+
+    }
+
+    public int swim(int distance) {
+        int actionCost = swimCost * distance;
+        if(actionCost > endurance) {
+            System.out.println("Животное "+name+" устало");
+            return -1;
+        }
+
+        endurance -= actionCost;
+        int time = distance/swimSpeed;
+        System.out.println("Животное "+name+" проплыло "+distance+" м за "+time+" c");
+
+        return time;
+    }
+
+    abstract int initSwimCost();
```

Comment:
Этот метод лишний

Reply:
Этот метод нужен для инициализации значения SwimCost у всех водоплавающих. Так как он абстрактный, то любое водоплавающее животное его реализует и это гарантирует, что всегда будет поле с названием SwimCost и оно всегда будет проинициализировано своим значением для конкретного животного.

Reply:
Опять же, идея не моя https://stackoverflow.com/questions/2613596/how-to-create-an-abstract-field
Использовал рекомендацию, которая показалась наиболее простой "using the Template pattern. In this way, you force all subclasses to implement the init() method, which, since it being called by the constructor, will assign the field for you."

Reply:
Там у человека хитрый случай, в данном решении такой подход не требуется

Reply:
Спорить не буду, вы профи, вам виднее